doc/Programming-guide/Your-first-Kaa-application/index.md

```diff
@@ -422,14 +422,147 @@ int main(void)
+ *
+ * @author Maksym Liashenko
+ * @see <a href="http://docs.kaaproject.org/display/KAA/Your+first+Kaa+application"> Documentation</a>
+ */
+public class FirstDemo {
+
+    private static final long DEFAULT_SAMPLE_TIME = 1000L;
+
+    private static final Logger LOG = LoggerFactory.getLogger(FirstDemo.class);
+    private static Timer timer = new Timer();
+    private static KaaClient kaaClient;
+
+
+    public static void main(String[] args) {
+        LOG.info("Kaa First Demo app starting!");
+        startKaaClient();
+
+        LOG.info("--= Press any key to exit =--");
+        try {
+            System.in.read();
+        } catch (IOException e) {
+            LOG.error("IOException has occurred: " + e.getMessage());
+        }
+        LOG.info("Stopping...");
 
-</div>
+        timer.cancel();
+        stopKaaClient();
+    }
 
-<div id="app-obj-c" class="tab-pane fade" markdown="1" >
+    private static void startKaaClient() {
+        /*
+         * Create the Kaa desktop context for the application.
+         */
+        DesktopKaaPlatformContext desktopKaaPlatformContext = new DesktopKaaPlatformContext();
+
+        /*
+         * Create a Kaa client and add a listener which displays the Kaa client
+         * configuration as soon as the Kaa client is started.
+         */
+        kaaClient = Kaa.newClient(desktopKaaPlatformContext, new SimpleKaaClientStateListener() {
+            @Override
+            public void onStarted() {
+                super.onStarted();
+                LOG.info("Kaa client started");
+
+                Configuration configuration = kaaClient.getConfiguration();
+                LOG.info("Default sample period: " + configuration.getSamplePeriod());
+
+                onKaaStarted(configuration.getSamplePeriod() * 1000L);
+
+            }
+        }, true);
+
+        kaaClient.setLogUploadStrategy(new RecordCountLogUploadStrategy(1));
+
+        /*
+         * Persist configuration in a local storage to avoid downloading it each
+         * time the Kaa client is started.
+         */
+        kaaClient.setConfigurationStorage(new SimpleConfigurationStorage(desktopKaaPlatformContext, "saved_config.cfg"));
+
+        kaaClient.addConfigurationListener(new ConfigurationListener() {
+            @Override
+            public void onConfigurationUpdate(Configuration configuration) {
+                LOG.info("Received configuration data. New sample period: " + configuration.getSamplePeriod());
+
+                onChangedConfiguration(configuration.getSamplePeriod() * 1000L);
+            }
+        });
+
+        /*
+         * Start the Kaa client and connect it to the Kaa server.
+         */
+        kaaClient.start();
+    }
 
-</div>
+    private static void addTemperatureRecord(DataCollection data) {
+        kaaClient.addLogRecord(data);
+    }
 
-</div>
+    private static void stopKaaClient() {
+        kaaClient.stop();
+    }
+
+    /*
+    * Retrieves current temperature.
```

Comment:
Add few words that we a re trying to simulate some temperature sensor (if it is not present in application description)